src/System.Net.Http/src/System/Net/Http/ByteArrayHelpers.cs

```diff
@@ -1,45 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System.Diagnostics;
-
-namespace System
-{
-    internal static class ByteArrayHelpers
-    {
-        // TODO #21395:
-        // Replace with the MemoryExtensions implementation of Equals once it's available
-        internal static bool EqualsOrdinalAsciiIgnoreCase(string left, ReadOnlySpan<byte> right)
```

Comment:
Any perf concerns with using the `ReadOnlySpan<char>` overload available now that will have the "IsAscii" check per iteration which is unnecessary here - https://github.com/dotnet/coreclr/blob/47c39edc2fbfbfbddca13ee1a47699ecaaaee204/src/mscorlib/shared/System/Globalization/CompareInfo.cs#L571?

Reply:
I don't see how the changes are valid. Casting a span of ASCII bytes to a span of chars will misinterpret then.

Reply:
Yep, you are definitely right. Here, two bytes would be considered one character rather than individual characters. Tests don't pass.

Reply:
You'd either need to Widen the `byte`s to `short`/`char` and compare
https://github.com/aspnet/KestrelHttpServer/blob/300453396a57cd14bcb297627b1507de83dc88ab/src/Kestrel.Core/Internal/Infrastructure/StringUtilities.cs#L12-L109
Or Narrow the `string` `char`s to `byte`s; which is probably dodgier (as would be discarding high byte)

Reply:
Can vectorize the casing with something like (not a very good version)
```csharp
public static void LowerCaseSIMD(byte[] data)
{
    var A = new Vector<byte>(65); // A
    var Z = new Vector<byte>(90); // Z

    for (var o = 0; o < data.Length - Vector<byte>.Count; o += Vector<byte>.Count)
    {
        var v = new Vector<byte>(data, o);

        v = Vector.ConditionalSelect(
            Vector.BitwiseAnd(
                Vector.GreaterThanOrEqual(v, A),
                Vector.LessThanOrEqual(v, Z)
            ),
            Vector.BitwiseOr(new Vector<byte>(0x20), v), // 0010 0000
            v
        );
```

Reply:
I was trying to see if there was some char to byte conversion happening up-stack, but the data is coming in as bytes... http://source.dot.net/#System.Net.Http/System/Net/Http/Managed/HttpConnection.cs,354

If we "Widen", won't we need to allocate a 2x temporary buffer?

Reply:
> If we "Widen", won't we need to allocate a 2x temporary buffer?

For the the vector part its two local `Vector<short>` variables; for shorter a `long`, `int` or `char`; no buffers needed